internal/concurrent/concurrent.go

```diff
@@ -39,14 +36,13 @@ func NewGroup(ctx context.Context) *Group {
 // Go runs func in backround.
 // Func should return when Group.Context is cancelled/done.
 func (g *Group) Go(f func()) {
-	g.Group.Go(func() error {
-		f()
-		return nil
-	})
+	g.group.Add(1)
+	go func() { f(); g.group.Done() }()
 }
 
 // Close cancels context and waits for threads to terminate.
 func (g *Group) Close() error {
```

Comment:
small comments:

1. can you not embed context and just expose `Done()`
2. rename `Close() error` to just `Wait()`

Reply:
First - sure.

Second - no, I'd prefer `concurrent.Group` to implement `bps.Subscription` (and I'd like to keep `Close() error` on `bps.Subscription` - seems more usual/common).

Reply:
the problem with `Close() error` is that you are inevitable checking the response of it. no error can happen here, we should not pretend that it might. we may end up bending backwards for no reason